pkg/util/node/node.go

```diff
@@ -103,46 +96,6 @@ func GetNodeHostIPs(node *v1.Node) ([]net.IP, error) {
 	return nodeIPs, nil
 }
 
-// GetNodeHostIP returns the provided node's "primary" IP; see GetNodeHostIPs for more details
-func GetNodeHostIP(node *v1.Node) (net.IP, error) {
-	ips, err := GetNodeHostIPs(node)
-	if err != nil {
-		return nil, err
-	}
-	// GetNodeHostIPs always returns at least one IP if it didn't return an error
-	return ips[0], nil
-}
-
-// GetNodeIP returns an IP (as with GetNodeHostIP) for the node with the provided name.
-// If required, it will wait for the node to be created.
-func GetNodeIP(client clientset.Interface, name string) net.IP {
```

Comment:
nit: 
`getNodeIPs` in `cmd/kube-proxy/app/server.go:653` is basicly a multi-stack version of `GetNodeHostIP` and it was copied and pasted from here. maybe we can move it back here and fix the unit-test, instead of remove this directly.

Reply:
For the record, from #119525, where this happened:

> FWIW it looks like GetNodeIP is only used by kube-proxy, and it's not obvious that any other part of k/k would have any need for it. (It's only useful for components that are running on a node but aren't kubelet and aren't running in a pod.)

(also, "and aren't cloud-controller-manager")

> So maybe GetNodeIPs should be in cmd/kube-proxy or pkg/proxy/util and you can submit a separate PR to just remove GetNodeIP later?

Apparently this never happened, but I think that's the right fix.

Reply:
Modified, thx